### DISCUSSION TITLE
What is the purpose of `description` YAML option?

### DISCUSSION THREAD

**Discussion by dense-set** [quarto-dev/quarto-cli#6055] [Q&A] [answered]
The [Quarto reference](https://quarto.org/docs/reference/projects/websites.html#section) for enlists the `description` option under the `website` key for website projects.

Where does the value stored in this option get used in the webpage? (The reference mentions this option among those that "affect website output", hence asking.)

**Comment by dragonstyle** [comment, marked as answer]
The description could appear in a couple of places:

1) It will be used as the `meta` description for the HTML page. Check the `<head>` tags to see this.

2) HTML title blocks may display the description as well.

HTH!

[edit below]

The above is true for the `description` key appearing in a document. You're asking more specifically about it appearing in `_quarto.yml` under `website` (sorry I didn't catch that on first reading).

When placed under the `website` key, it looks like we use that for the description that appears within any RSS feed generated for the website (if no description is available on the page which generated the RSS feed).

**Reply by dense-set** [reply]
But I am not able to access it by the shortcode `{{< meta website.description >}}` (also see #6049), which yields:

<img width="146" alt="Screenshot 2023-06-30 at 10 35 41 PM" src="https://github.com/quarto-dev/quarto-cli/assets/102873841/6fc13bef-af6e-429d-8bdb-b6d8e117876a">

**Reply by dragonstyle** [reply]
Project metadata isn’t considered a part of document metadata currently, so keys such as book and website aren’t available at the document level for the meta shortcode right now…

**Reply by dense-set** [reply]
But I have the following observation:

`_quarto.yml`:
````yml
project:
  type: website

website:
  title: "my website"

format:
  html:
    theme: cosmo
    css: styles.css
    toc: true
````

`index.qmd`:
````qmd
---
title: "index title"
---

{{< meta website.title >}}
````

This doesn't give any error (unlike `{{< meta website.description >}}`), and instead outputs the following:
<img width="598" alt="Screenshot 2023-07-01 at 7 31 42 PM" src="https://github.com/quarto-dev/quarto-cli/assets/102873841/e3a0f5b5-1fda-45cc-9b1a-ba742da6f9e0">

This suggests that the variable `website.title` _is_ accessible at the document level, and is assigned to the value of document variable `title`.

**Reply by dragonstyle** [reply]
Thanks for finding this - you are correct that `website.title` was resolving (though it shouldn't have). Due to a bug, we were actually resolving the `title` against the title in the document metadata (rather than the website title). I have a fix for that issue on the way.

**Reply by mcanouil** [reply]
@dragonstyle FYI the issue about shortcode was already discussed elsewhere: https://github.com/quarto-dev/quarto-cli/discussions/6049#discussioncomment-6329278

**Reply by dragonstyle** [reply]
Thanks! There are really two issue in play here:

1) Can you reference `website` metadata (currently no). I haven't changed anything about this.

2) Are dereferenced metadata keys properly resolved (currently no, if the root key like `website` wasn't present in the metadata, we were just ignoring it, and resolving the next level key against the root metadata). I did fix this- `{{< meta website.title >}}` will now warn and fail to resolve rather than resolving again the document title.
